src/locutus/api/table_mappings.py

```diff
@@ -12,26 +12,44 @@
 class TableMappings(Resource):
     @classmethod
     def get_mappings(cls, id):
-        table = Table.get(id)
-        term = table.terminology.dereference()
-
-        response = {
-            "terminology": {
-                "Reference": f"Terminology/{term.id}",
-            },
-            "codes": [],
-        }
-        mappings = term.mappings()
-
-        for code in mappings:
-            mapping = {"code": code, "mappings": []}
-            for codingmapping in mappings[code]:
-                mapping["mappings"].append(codingmapping.to_dict())
-
-            response["codes"].append(mapping)
-
-        return response
+        user_input_param = request.args.get("user_input", default=None)
+        editor_param = request.args.get("user", default=None)
+        try:
+            editor = get_editor(body=None, editor=editor_param)
+            if user_input_param is not None and editor is None:
+                raise LackingUserID(editor)
+            
+            table = Table.get(id)
+            term = table.terminology.dereference()
 
+            response = {
+                "terminology": {
+                    "Reference": f"Terminology/{term.id}",
+                },
+                "codes": [],
+            }
+            mappings = term.mappings()
+
+            for code in mappings:
+                mapping = {"code": code, "mappings": []}
+                for codingmapping in mappings.get(code, []):
+                    if user_input_param:
+                        user_input_data = (
+                            MappingUserInputModel.generate_mapping_user_input(
+                                term.id, code, codingmapping.code, editor

```

Comment:
The only difference from Table and equivalent Terminology functions is that the term.id needs to be specified here, instead of id(which is the Table id in this instance).